src/Symfony/Component/Messenger/Transport/Receiver/DisconnectedReceiverInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Receiver;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\TransportException;
+
+/**
+ * Used when a receiver does not keep persistent connection with the broker and have to explicitly tell when messages
+ * are not proceeded and have to be re-distributed.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+interface DisconnectedReceiverInterface
+{
+    /**
+     * Sends back prefetched messages into the queue.
+     *
+     * @throws TransportException If there is an issue communicating with the transport
+     */
+    public function requeuePrefetchedMessages(): void;
```

Comment:
The only reason is because of the pre-fetch. What's the actual use-case of this method? I can only see one: the consumer wants to safely shutdown. In this use-case, it just needs not to fetch new ones and go through its pre-fetched. In the event of the consumer having an incident, it would rely on the queue's timeout to acknowledge... can't we just rely on this?

Reply:
> it just needs not to fetch new ones and go through its pre-fetched

1. if the consumer stops because it reach a "maximum message proceeed", it would be weird to handle few more messages. 
2. by design the consumer (not aware of internal state of transports) asks to the transport to retrieves messages one by one, implementing such "flush without fetching new-one" will add more complexity IMO

 > can't we just rely on this [queue when timeout]?

Waiting for timeout may be acceptable in case of accident, but not in "normal" situation:
ie: when the worker is configured to restart every X messages or X seconds, it's may be not acceptable to have a 9 messages in timeout every X messages/seconds

Reply:
removed and replaced by `__destruct` in connection